osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -860,9 +860,13 @@ private bool transferRulesetValue()
             Logger.Log($"decoupled ruleset transferred (\"{decoupledRuleset.Value}\" -> \"{Ruleset.Value}\")");
             rulesetNoDebounce = decoupledRuleset.Value = Ruleset.Value;
 
-            // if we have a pending filter operation, we want to run it now.
-            // it could change selection (ie. if the ruleset has been changed).
-            Carousel.FlushPendingFilterOperations();
+            if (Carousel.IsLoaded)
```

Comment:
I think I would prefer this inside the method as proposed. Not that I expect this to be a problem anywhere else (looks like it can't be to me), is just weird that a child's state is checked at a higher level.